gateway-js/src/legacy/serviceListShim.ts

```diff
@@ -0,0 +1,142 @@
+import {
+  composeAndValidate,
+  compositionHasErrors,
+  ServiceDefinition,
+} from '@apollo/federation';
+import CallableInstance from 'callable-instance';
+import { HeadersInit } from 'node-fetch';
+import {
+  GraphQLDataSource,
+  RemoteGraphQLDataSource,
+  ServiceEndpointDefinition,
+  SupergraphSdlUpdateFunction,
+} from '..';
+import { SupergraphSdlHookOptions, SupergraphSdlHookReturn } from '../config';
+import {
+  getServiceDefinitionsFromRemoteEndpoint,
+  Service,
+} from '../loadServicesFromRemoteEndpoint';
+
+export interface ServiceListShimOptions {
+  serviceList: ServiceEndpointDefinition[];
+  introspectionHeaders?:
+    | HeadersInit
+    | ((
+        service: ServiceEndpointDefinition,
+      ) => Promise<HeadersInit> | HeadersInit);
+  buildService?: (definition: ServiceEndpointDefinition) => GraphQLDataSource;
+  pollIntervalInMs?: number;
+}
+
+export class ServiceListShim extends CallableInstance<
```

Comment:
If we think this is going to be a reasonably popular thing for people to choose to use even in new code (vs just something that is purely a migration aid) maybe give it a more descriptive name like "introspect and compose"?

Reply:
I think I'm going to keep this under `legacy`. I think this is a totally valid use case but I also think the hope is that we nudge users to a rover-based workflow in the future. cc @abernix if you have any additional thoughts here.

Reply:
Moved to top-level `IntrospectAndCompose` folder